packages/@ember/-internals/glimmer/lib/helpers/fn.ts

```diff
@@ -1,43 +1,11 @@
-import { HAS_NATIVE_PROXY } from '@ember/-internals/utils';
 import { assert } from '@ember/debug';
-import { DEBUG } from '@glimmer/env';
 import { Arguments, VM } from '@glimmer/runtime';
 import { ICapturedArguments } from '@glimmer/runtime/dist/types/lib/vm/arguments';
 import { Opaque } from '@glimmer/util';
 import { InternalHelperReference } from '../utils/references';
+import buildUntouchableThis from '../utils/untouchable-this';
 
-let context: any = null;
-if (DEBUG && HAS_NATIVE_PROXY) {
-  let assertOnProperty = (property: string | number | symbol) => {
-    assert(
-      `You accessed \`this.${String(
-        property
-      )}\` from a function passed to the \`fn\` helper, but the function itself was not bound to a valid \`this\` context. Consider updating to usage of \`@action\`.`
-    );
-  };
-
-  context = new Proxy(
-    {},
-    {
-      get(_target: {}, property: string | symbol) {
-        assertOnProperty(property);
-      },
-
-      set(_target: {}, property: string | symbol) {
-        assertOnProperty(property);
-
-        return false;
-      },
-
-      has(_target: {}, property: string | symbol) {
-        assertOnProperty(property);
-
-        return false;
-      },
-    }
-  );
-}
-
+const context = buildUntouchableThis('`fn` helper');
```

Comment:
![Can't touch this](https://media.giphy.com/media/hFIkt5s7MrzI4/giphy.gif)

Reply:
yaaaaas